include/RAJA/pattern/launch/launch_core.hpp

```diff
@@ -258,12 +261,14 @@ void launch(ExecPlace place, const LaunchParams &params, const char *kernel_name
   //Forward to single policy launch API - simplifies testing of plugins
   switch (place) {
     case ExecPlace::HOST: {
-      launch<LaunchPolicy<typename POLICY_LIST::host_policy_t>>(params, kernel_name, body);
+      using Res = typename resources::get_resource<typename POLICY_LIST::host_policy_t>::type;
+      launch<LaunchPolicy<typename POLICY_LIST::host_policy_t>>(Res::get_default(), params, kernel_name, body);
       break;
     }
 #ifdef RAJA_DEVICE_ACTIVE
   case ExecPlace::DEVICE: {
-      launch<LaunchPolicy<typename POLICY_LIST::device_policy_t>>(params, kernel_name, body);
+      using Res = typename resources::get_resource<typename POLICY_LIST::device_policy_t>::type;
+      launch<LaunchPolicy<typename POLICY_LIST::device_policy_t>>(Res::get_default(), params, kernel_name, body);
```

Comment:
@MrBurmark What do you think of generating the default resource here and passing it in?

Reply:
Looks good to me.

Reply:
It looks like this place let you de-duplicate the cuda launch backend, nice. The rest are coming?

Reply:
I see three front-ends here policy, place, and resource.
I see that the place version already calls through to one of the other front ends.
Is the plan to eventually have all of these three funnel through a non-generic resource front-end so the plugin stuff can be done just once?
I think we discovered previously that we will need to update camp to make this work for all cases as a non-generic event proxy can't be turned into a generic event proxy.

Reply:
Yes, at this point it is exploratory (just trying to see what could work) -- I'll move forward with the other backends. That is a good point about the necessary update to camp. What we could do is clean up the backend specific exec methods in this PR and then fix up the launch calls in a different PR in which we work with camp to our address the project needs.

Reply:
Just pushed up the omp, sycl, and hip fixes. Just need to build the sycl version locally to test (will be able to do that next week).

Reply:
The one thing that I think still needs to be done here is to have the backends accept the typed resource instead of the type-erased resource. That can be done in another PR since that is at this point an internal detail. I think the reason that wasn't done initially way the difficulty of converting a typed event proxy into a type-erased event proxy which requires an update in camp. I think we should merge this PR, add support for the event proxy conversions to camp, then update this so we can use typed resources in the backends before the next release. What do you think @artv3?

Reply:
yes, sounds great to me.